lib/collection/src/shards/transfer/shard_transfer.rs

```diff
@@ -268,12 +269,24 @@ async fn transfer_snapshot(
         .create_shard_snapshot(snapshots_path, collection_name, shard_id, temp_dir)
         .await?;
 
+    let snapshot_temp_path = shard_holder_read
+        .get_shard_snapshot_path(snapshots_path, shard_id, &snapshot_description.name)
+        .await
+        .map(TempPath::from_path)
+        .map_err(|err| {
+            log::error!(
+                "Failed to determine snapshot path, will not delete file after recovery: {err}"
+            );
+        })
+        .ok();
```

Comment:
Should we just bail here?

```suggestion
    let snapshot_temp_path = shard_holder_read
        .get_shard_snapshot_path(snapshots_path, shard_id, &snapshot_description.name)
        .await
        .map(TempPath::from_path)
        .map_err(|err| {
            log::error!(
                "Failed to determine snapshot path, will not delete file after recovery: {err}"
            );
        })?;
```

Reply:
IIRC, `get_shard_snapshot_path` only fails if `shard_id` does not exist or is not a local shard, in which case we _probably_ want to cancel the shard transfer anyway.